bindings/cs/rl.net/AsyncSender.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Rl.Net.Native;
+
+namespace Rl.Net
+{
+    public abstract class AsyncSender : ISender
+    {
+        private ErrorCallback errorCallback;
+
+        public AsyncSender(ErrorCallback callback)
```

Comment:
I will add the event as an indirection mechanism, though it is odd to expose an event on a class, then only expect it to be used to raise events, rather than subscribe to them.